navigation/navigation-runtime-ktx/src/main/java/androidx/navigation/NavController.kt

```diff
@@ -26,3 +31,19 @@ public inline fun NavController.createGraph(
     @IdRes startDestination: Int,
     builder: NavGraphBuilder.() -> Unit
 ): NavGraph = navigatorProvider.navigation(id, startDestination, builder)
+
+/**
+ * Creates and returns a [Flow] that will emit the currently active [NavBackStackEntry] whenever
+ * it changes. If there is no active [NavBackStackEntry], no item will be emitted.
+ */
+@Suppress("EXPERIMENTAL_API_USAGE")
+public fun NavController.getBackStackEntryFlow(): Flow<NavBackStackEntry> = callbackFlow {
+    val listener = NavController.OnDestinationChangedListener { controller, _, _ ->
+        controller.currentBackStackEntry?.let(::sendBlocking)
```

Comment:
would it be better to use second argument instead?  given that it is non null it will make things as easy as `sendBlocking(destination)`

another corner case question: sendBlocking can theoretically block on the main thread, when there is slow consumer and a lot of actions here. Though it seems like fairly theoretical concern

Reply:
Would be possible, but then it would be a `Flow<NavDestination>` instead of a `Flow<NavBackStackEntry>`, which is not what we want I guess (as `NavBackStackEntry` is the class that gives access to things like `ViewModelStore` and the `SavedStateRegistry`). But maybe it would make sense to have both, a `backStackEntryFlow` and a `destinationFlow`?

Valid point, I also wasn't sure what's best to use here. The other option would be to use `offer`, which could lead to lost items it the consumer is slow.

Reply:
We are specifically looking for the `NavBackStackEntry` here, not `NavDestination`.